_posts/2018-11-15-using-vault-with-strimzi.md

```diff
@@ -0,0 +1,205 @@
+> Some secrets engines simply store and read data - like encrypted Redis/Memcached. Other secrets engines connect to other services and generate dynamic credentials on demand. Other secrets engines provide encryption as a service, totp [time-based one-time password] generation, certificates, and much more.
+
+It is this last point, certificates, which we are going to be looking at. Specifically, we will be using the [Vault PKI Secrets Engine](https://www.vaultproject.io/docs/secrets/pki/index.html) to create a CA, which we will then install instead of using CA certificates and private keys generated by the [Strimzi Cluster Operator](http://strimzi.io/docs/master/#cluster-operator-str).
+
+# Securing a Kafka cluster on Kubernetes
+
+## The `incubator` Kafka charts
+
+Like me, you may have tried configuring TLS for Kafka using the `incubator` Helm charts (you can find them [here](https://github.com/helm/charts/tree/master/incubator/kafka)), and realised that in fact this is _far from trivial_. I had initially configured an `initContainer` to get certificates from the Vault PKI backend using the Vault CLI, which did indeed work. However, in addition to the startup scripts in the main broker container and another custom `initContainer` I had configured to set up rack awareness, this caused the the pod startup times to become quite long!
+
+At this point I had configured the brokers and clients with TLS, but what about the Zookeepers? Zookeeper does not support TLS, which is problematic if, for example, you are a financial organisation streaming sensitive data through your Kafka cluster. Even if you're not a fintech company, but you're a sensible individual with security concerns, this might be a deal breaker for you!
+
+## Strimzi & Security
+
+Straight out of the box, [Strimzi provides all of this for you](http://strimzi.io/docs/master/#security-str):
+
+> Strimzi supports encrypted communication between the Kafka and Strimzi components using the TLS protocol. Communication between Kafka brokers (interbroker communication), between Zookeeper nodes (internodal communication), and between these and the Strimzi operators is always encrypted. Communication between Kafka clients and Kafka brokers is encrypted according to how the cluster is configured. For the Kafka and Strimzi components, TLS certificates are also used for authentication.
+> The Cluster Operator automatically sets up TLS certificates to enable encryption and authentication within your cluster. It also sets up other TLS certificates if you want to enable encryption or TLS authentication between Kafka brokers and clients.
+
+By default, Strimzi generates its own CA and stores the CA certificate and private key as Kubernetes secrets. Using these two secrets, Strimzi generates certificates for all of the components in the cluster, which in turn ensures all communication is encrypted!
```

Comment:
You probably already know this, but there are actually two CAs. One (the "cluster CA") is used for all internal communication within the cluster while the other (the "clients CA") is used for client applications. For the purposes of this post they're sufficiently similar that it doesn't concretely affect what you're doing, but it's a point worth making because if the user has a requirement for using Vault for one of the CAs it's likely the same requirement will apply to the other one too.

Reply:
I have clarified this in that paragraph now!